contracts/Hub.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title Hub Contract that coordinates Storage, Fund and Minter contract.
+ * @dev The contract can mint new tokens, check the collateral ratio of addresses
+ * and liquidate addresses if the collateral ratio falls below 150%
+ * @author The Everest team.
+ */
+
+contract Hub is Ownable {
+    // Contract addresses
+    Minter private minterContract;
+    Fund private fundContract;
+    Storage private storageContract;
+
+    address[] private authorizedAddresses;
+    mapping(string => address) private tickersymbolToAddress;
+
+
+    modifier onlyAuthorizedAddresses() {
+        bool isAuthorized = false;
+        for(uint i = 0; i < authorizedAddresses.length; i++) {
+            if(authorizedAddresses[i] == msg.sender) {
+                isAuthorized = true;
+            }
+        }
+        require(isAuthorized);
+        _;
+    }
+
+    constructor() {
+    }
+
+    /**
+     * @notice Add an address to the list of authorized addresses
+     * @param addr address that gets added 
+     */
+    function addAuthorizedAddress(address addr) public onlyOwner{
+        authorizedAddresses.push(addr);
+    }
+
+    /**
+     * @notice updates the address asssociated with an synthAsset
+     * @param tickerSymbol identifier of synthAsset
+     * @param addr new contract address of synthAsset 
+     */
+    function updateContractProxy(string memory tickerSymbol, address addr) public onlyAuthorizedAddresses{
+        tickersymbolToAddress[tickerSymbol] = addr; 
+    }
+
+    /**
+     * @notice mints synthAssets to a specific address
+     * @param receiver address to which token gets minted
+     * @param amount amount of token that gets minted
+     * @param tickerSymbol identifier of which token gets minted
+     */
+    function mintToken(address receiver, uint256 amount, string memory tickerSymbol) public onlyAuthorizedAddresses{
+        //TODO: mintToken for a specific synthAsset. 
+        //How do we interact with contracts we don't have the interface for
+        //-> every synthAsset contract
+        //maybe heritage??
+        minterContract.mint(receiver, amount); 
+    }
+
+    /**
+     * @notice sets the Minter contract address
+     * @param _minterAddress address of Minter contract
+     */
+    function setMinterContract(address _minterAddress) public onlyOwner{
+        minterContract = Minter(_minterAddress);
+    }
```

Comment:
we don't need this function anymore, `tickersymbolToSynthContract` replaces field `minterContract`